website/docs/consumer.md

```diff
@@ -104,3 +104,459 @@ Use the [can-i-deploy](https://github.com/pact-foundation/pact_broker/wiki/Provi
+      expect(products).toEqual([{ id: "1", name: "Product 1" }]);
+    });
+});
+```
+
+**❌ Bad:**
+```ts
+test("calculates total cart value", () => {
+  const cart = new Cart();
+  cart.addItem({ price: 10 });
+  expect(cart.total()).toBe(10); // functional test, not contract test
+});
+```
+
+### 🧭 3. Clarity and Maintainability
+
+#### ✅ Rule: Use provider states for clarity and maintainability
+State setup should be explicit.
+
+**Explanation:**
+This improves test clarity and helps the provider team replicate test conditions during verification.
+
+**✅ Good:**
+```ts
+.given("product with ID 1 exists")
+.uponReceiving("a request for product with ID 1")
+.withRequest("GET", "/products/1")
+.willRespondWith(200, ...)
+```
+
+**❌ Bad:**
+```ts
+.uponReceiving("a request for product with ID 1")
+.withRequest("GET", "/products/1")
+.willRespondWith(200, ...)
+// No provider state set — makes provider testing harder, reduces comprehension
+```
+
+#### ✅ Rule: Use matchers to reduce test fragility
+Avoid asserting on exact values unless necessary.
+
+**Explanation:**
+Using matchers like `like()` or `eachLike()` ensures the contract is flexible and tolerant to non-breaking changes.
+
+**✅ Good:**
+```ts
+eachLike({ id: like("123"), name: like("Product 1") })
+```
+
+**❌ Bad:**
+```ts
+[ { id: "123", name: "Product 1" } ]  // Makes the provider tests more brittle and coupled
+```
+
+#### 🧱 Avoid over-constraining the provider
+
+Only assert on formats or constraints that your consumer actually depends on.
+
+**Explanation**:  
+Avoid over-constraining the provider by specifying overly strict formats (e.g. using regex patterns) unless your consumer truly relies on them. While you might know how the provider currently behaves, asserting on things your consumer doesn’t need — like a specific ID format — can create unnecessary fragility. This can lead to false test failures if the provider changes in ways that don’t affect your consumer. Keep your assertions just strict enough to ensure your consumer continues to work as expected — and no stricter.
+
+
+✅ OK:
+```json
+{
+  "customerId": "ABC-123"
+}
+```
+
+❌ Over-constrained:
+```ts
+{
+  "customerId": regex("[A-Z]{3}-\d{3}", "ABC-123")  // forces provider to constrain to a regex
+}
+```
+
+Instead, keep your assertions just strict enough to ensure your consumer works as expected — and no stricter. This helps the provider evolve safely without false test failures.
+
+#### ✅ Rule: Avoid duplicate interaction descriptions
+Make each interaction description unique.
+
+**Explanation:**
+The Pact Broker de-duplicates interactions based on description + provider state. Re-using the same description can result in dropped tests.
+
+**✅ Good:**
+```ts
+.given("there are active products in the catalog")
+.uponReceiving("a request for a list of products")
+```
+
+**❌ Bad:**
+```ts
+.uponReceiving("a request for a list of products") // no provider state
```

Comment:
maybe expand this to contain two tests with same description, one expecting a 200 / 404 or empty list